src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Domain\Settings\Model\ValueObject;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ConnectionType
+{
+    const DEFAULT_TYPE = 'default';
+
+    private string $type;
+
+    public function __construct(?string $type = self::DEFAULT_TYPE)
+    {
+        $type = $type ?? self::DEFAULT_TYPE;
+
+        if (empty($type)) {
+            throw new \InvalidArgumentException('akeneo_connectivity.connection.connection.constraint.type.required');
+        }
+
+        if (mb_strlen($type) > 30) {
+            throw new \InvalidArgumentException('akeneo_connectivity.connection.connection.constraint.type.too_long');
+        }
+
+        $this->type = $type;
+    }
+
+    public function __toString()
```

Comment:
```suggestion
    public function __toString(): string
```
I think it is still required even if you use the magical function.